### DISCUSSION TITLE
How to use multiple namespaces and service accounts?

### DISCUSSION THREAD

**Discussion by enter-marlah** [apache/camel-karavan#1158] [Q&A] [answered]
Hello,

I would like to deploy each project to different namespace using a different service account in Kubernetes. if possible I'd like to use project id as namespace and service account name. Is this possible?

I can deploy project to different namespace by hardcoding the namespace name in kubernetes-build.sh but I guess this is not the correct way since the deployment is not visible in the Karavan UI deployment tab.

Any help would be greatly appreciated!

Thanks, Marko

**Comment by mgubaidullin** [comment]
Current implementation is designed to deploy all projects in the same namespace.
What is the reason that you need to deploy each microservice in separate namespace?

**Reply by enter-marlah** [reply]
We are running several integration for our customers and want to separate each customer into own namespace. Can you suggest a better approach for housekeeping?

**Reply by mgubaidullin** [reply, marked as answer]
As for now you could install karavan per customer

**Reply by enter-marlah** [reply]
Thansk for your help!
